### PR TITLE
Schedule api

### PR DESCRIPTION
**Pull request by setmao**
## Types of changes
**Thanks for sending a pull request! Please fill in the following content to let us know better about this change.**
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [X] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
Response format
```JSON
{
    "data": [
        {
            "date": "2021-10-02",
            "name": "第 1 天",
            "rooms": [
                "1-r3",
                "2-all",
                "3-r012",
                "4-r0",
                "5-r1",
                "6-r2"
            ],
            "slots": {
                "2-all": [
                    {
                        "event_id": 146,
                        "event_type": "custom",
                        "title": "Registration & Breakfast",
                        "speakers": [],
                        "begin_time": "2021-10-02 00:30:00",
                        "end_time": "2021-10-02 00:50:00",
                        "is_remote": false,
                        "recording_policy": true,
                        "language": "",
                        "python_level": "",
                        "break_event": false
                    },
                    ...
                ],
                ...
            },
            "timeline": {
                "begin": "2021-10-02 00:30:00",
                "end": "2021-10-02 12:00:00"
            }
        },
        ...
    ]
}
```

## Steps to Test This Pull Request
1. API endpoint: http://localhost:8000/api/events/schedule/

## Expected behavior
Should see the response with right data format.

### CODE REVIEW

File: src/pycontw2016/settings/base.py [hunk 1]
```diff
@@ -323,8 +323,8 @@ def node_bin(name):
 )
 
 EVENTS_DAY_NAMES = collections.OrderedDict([
-    (datetime.date(2020, 9, 5), _('Day 1')),
```
Comment (mattwang44) [inline src/pycontw2016/settings/base.py:326]: already done in #1030. may need to rebase your branch to the latest master

Comment (setmao) [reply src/pycontw2016/settings/base.py]: Fixed

File: src/events/api/views.py [hunk 2]
```diff
@@ -62,9 +62,110 @@ def get(self, request):
...
+    def title(self) -> str:
+        if isinstance(self.obj, KeynoteEvent):
+            return self.obj.session_title
+        elif isinstance(self.obj, (ProposedTalkEvent, ProposedTutorialEvent)):
+            return self.obj.proposal.title
+        else:
+            return self.obj.title
+
+    @property
+    def speakers(self) -> str:
```
Comment (mattwang44) [inline src/events/api/views.py:105]: return as a list of speaker names

File: src/events/api/views.py [hunk 3]
```diff
@@ -62,9 +62,110 @@ def get(self, request):
...
+    @property
+    def break_event(self) -> bool:
+        if isinstance(self.obj, CustomEvent):
+            return self.obj.break_event
+        else:
+            return False
+
+    @property
+    def language(self) -> str:
+        if isinstance(self.obj, (ProposedTalkEvent, ProposedTutorialEvent)):
```
Comment (mattwang44) [inline src/events/api/views.py:150]: handle for SponsoredEvent

File: src/events/api/views.py [hunk 4]
```diff
@@ -62,9 +62,110 @@ def get(self, request):
...
+            return self.obj.proposal.language
+        else:
+            return ''
+
+    @property
+    def python_level(self) -> str:
+        if isinstance(self.obj, (ProposedTalkEvent, ProposedTutorialEvent)):
```
Comment (mattwang44) [inline src/events/api/views.py:157]: handle SponsoredEvent

File: src/events/api/views.py (hunk 2 above)
Comment (setmao) [reply src/events/api/views.py]: Fixed, now speakers will return a list of str

File: src/events/api/views.py (hunk 3 above)
Comment (setmao) [reply src/events/api/views.py:150]: Fixed

File: src/events/api/views.py (hunk 4 above)
Comment (setmao) [reply src/events/api/views.py:159]: Fixed

File: src/events/api/views.py [hunk 5]
```diff
@@ -176,8 +180,8 @@ def display(self):
 
 
 class ScheduleAPIView(APIView):
-    authentication_classes = [TokenAuthentication]
-    permission_classes = [IsAuthenticated]
+    # authentication_classes = [TokenAuthentication]
+    # permission_classes = [IsAuthenticated]
```
Comment (mattwang44) [inline src/events/api/views.py:184]: Uncomment

Comment (setmao) [reply src/events/api/views.py]: Fixed

**Pull request merged**